### PR TITLE
set default useInstanceMetadataHostname true for external-aws

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/43175

Setting default value for the field `useInstanceMetadataHostname` when using `external-aws`. Docs and release notes would be updated to note it'll have to be disabled when users are passing custom node names for custom clusters. For all other usecases, the field will be true so that external cloud provider can find the node without `providerID`.

### CODE REVIEW

File: pkg/api/norman/store/cluster/cluster_store.go
Comment (jiaqiluo) [inline pkg/api/norman/store/cluster/cluster_store.go:1382]: We can simplify the blob `if ok && rkeConfig != nil {`:
``` go
	if ok && rkeConfig != nil {
		cloudProviderName := convert.ToString(values.GetValueN(data, "rancherKubernetesEngineConfig", "cloudProvider", "name"))
		_, ok := values.GetValue(data, "rancherKubernetesEngineConfig", "cloudProvider", "useInstanceMetadataHostname")
		if !ok {
			var val bool
			switch cloudProviderName {
			case k8s.AWSCloudProvider:
				// set default false for aws cloud provider
				val = false
			case k8s.ExternalAWSCloudProviderName:
				// set default true for external aws cloud provider
				val = true
			}
			values.PutValue(data, val, "rancherKubernetesEngineConfig", "cloudProvider", "useInstanceMetadataHostname")
		}
	}
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: pkg/api/norman/store/cluster/cluster_store.go
Comment (kinarashah) [reply pkg/api/norman/store/cluster/cluster_store.go]: cleaned it up

**Review by jiaqiluo — APPROVED** [review]
LGTM

**Review by jakefhyde — APPROVED** [review]

File: pkg/api/norman/store/cluster/cluster_store.go
Comment (snasovich) [inline pkg/api/norman/store/cluster/cluster_store.go:1377]:
```suggestion
				// set default true for external-aws cloud provider
```

**Review by snasovich — APPROVED** [review]
The only change is trivial comment fix (https://github.com/rancher/rancher/compare/98e992766ab865ffe2b68ff928f4a820447785fa..09f53e8cda5f3e90ccac6721d83db22bd714247f) so merging without additional approval or waiting for CI.

**Pull request merged**
